Makefile

```diff
@@ -79,10 +70,10 @@ lima-template: download
 	rm $(OUTDIR)/lima-template/*.yaml.bak
 
 .PHONY: lima-socket-vmnet
-lima-socket-vmnet: download.socket_vmnet
-	mkdir -p $(SOCKET_VMNET_DEPDIR)
-	tar -zvxf "$(DOWNLOAD_DIR)/$(SOCKET_VMNET_BASENAME)" -C $(SOCKET_VMNET_DEPDIR) --strip-component=1
-	cd $(SOCKET_VMNET_DEPDIR) && $(MAKE) PREFIX=$(SOCKET_VMNET_TEMP_PREFIX) install.bin
+lima-socket-vmnet:
+	git submodule update --init --recursive src/socket_vmnet
+	cd src/socket_vmnet && git clean -f -d
+	cd src/socket_vmnet && PREFIX=$(SOCKET_VMNET_TEMP_PREFIX) $(MAKE) install.bin
```

Comment:
This would create the binary in the submodule directory. In finch cli, `socket_vmnet` is referenced from [/dependencies/lima-socket_vmnet](https://github.com/runfinch/finch/blob/main/pkg/dependency/vmnet/binaries.go#L56). We can probably refactor the cli later and just move the binary to `downloads/dependencies/lima-socket_vmnet`, wdyt?

Reply:
You're right that it creates a binary in the submodule directory, but the install.bin target of the Makefile moves it to the expected directory (e.g. `./_output/dependencies/lima-socket_vmnet/opt/finch/bin/socket_vmnet`)

Reply:
My bad, just ran tested the `lima-socket-vmnet` on my local, it works fine.